src/main/java/org/folio/service/event/handler/processor/ServicePointDeleteProcessorForRequestPolicy.java

```diff
@@ -0,0 +1,96 @@
+import org.apache.logging.log4j.Logger;
+import org.folio.persist.RequestPolicyRepository;
+import org.folio.rest.jaxrs.model.AllowedServicePoints;
+import org.folio.rest.jaxrs.model.RequestPolicy;
+import org.folio.rest.jaxrs.model.RequestType;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+
+public class ServicePointDeleteProcessorForRequestPolicy
+  extends DeleteEventProcessor<RequestPolicy> {
+
+  private static final Logger log = LogManager.getLogger(ServicePointDeleteProcessorForRequestPolicy.class);
+
+  private final Context context;
+
+  public ServicePointDeleteProcessorForRequestPolicy(Context context) {
+    super(INVENTORY_SERVICE_POINT_DELETED);
+    this.context = context;
+  }
+
+  @Override
+  protected List<Change<RequestPolicy>> collectRelevantChanges(JsonObject payload) {
+    log.debug("collectRelevantChanges:: payload: {}", payload);
+
+    JsonObject oldObject = payload.getJsonObject("old");
+
+    List<Change<RequestPolicy>> changes = new ArrayList<>();
+    String deletedServicePointId = oldObject.getString("id");
+
+    changes.add(new Change<>(requestPolicy -> {
+      AllowedServicePoints allowedServicePoints = requestPolicy.getAllowedServicePoints();
+
+      removeAllowedServicePoint(deletedServicePointId, requestPolicy, RequestType.HOLD,
+        allowedServicePoints::getHold, allowedServicePoints::setHold);
+      removeAllowedServicePoint(deletedServicePointId, requestPolicy, RequestType.PAGE,
+        allowedServicePoints::getPage, allowedServicePoints::setPage);
+      removeAllowedServicePoint(deletedServicePointId, requestPolicy, RequestType.RECALL,
+        allowedServicePoints::getRecall, allowedServicePoints::setRecall);
+    }));
+
+    return changes;
+  }
+
+  private void removeAllowedServicePoint(String deletedServicePointId, RequestPolicy requestPolicy,
+    RequestType requestType, Supplier<Set<String>> getAllowedServicePointsSupplier,
+    Consumer<Set<String>> setAllowedServicePointsConsumer) {
+
+    log.debug("removeAllowedServicePoint:: deletedServicePointId={}, requestPolicy={}, " +
+        "requestType={}", deletedServicePointId, requestPolicy, requestType);
+
+    Set<String> allowedServicePoints = getAllowedServicePointsSupplier.get();
+
+    if (allowedServicePoints == null) {
+      log.info("removeAllowedServicePoint:: allowed service points missing for type {}",
+        requestType);
+      return;
+    }
+
+    allowedServicePoints.remove(deletedServicePointId);
+    if (allowedServicePoints.isEmpty()) {
+      log.info("removeAllowedServicePoint:: request policy ID={}: 0 allowed service point for {} " +
+        "type, removing it from the policy", requestPolicy::getId, () -> requestType);
+
+      setAllowedServicePointsConsumer.accept(null);
+      requestPolicy.getRequestTypes().remove(requestType);
+    }
+  }
+
+  @Override
+  protected Future<List<RequestPolicy>> applyChanges(List<Change<RequestPolicy>> changes,
```

Comment:
This method looks very similar in all processors:
- extract old object from payload
- instantiate repository
- fetch old object by id
- apply db updates

I think if we inject repository via constructor and extract 3d step (fetch objects by id) into an abstract method, we can make `applyChanges()` non-abstract and move it into superclass. This way you won't have to pull headers along  all the way.